tests/test_send_telemetry_appinsights.py

```diff
@@ -0,0 +1,96 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import sys
+import logging
+from importlib import reload
+from unittest.mock import patch
+import pytest
+from featuremanagement import EvaluationEvent, FeatureFlag, Variant, VaraintAssignmentReason
+import featuremanagement.azuremonitor._send_telemetry
+
+
+@pytest.mark.usefixtures("caplog")
+class TestSendTelemetryAppinsights:
+
+    def test_send_telemetry_appinsights(self):
```

Comment:
I think we'd better have integration test like [this](https://github.com/microsoft/FeatureManagement-Dotnet/blob/preview/tests/Tests.FeatureManagement/FeatureManagementTest.cs#L1336)

We call IsEnabled/GetVariant on a feature flag and we check the telemetry.

Reply:
These are unit tests, not integration tests.

Reply:
Can we add some integration test?

Reply:
I don't think any of our libraries besides Spring do actually integration tests. We would need to setup stores and resources to collect the telemetry.

Reply:
Ok. I am just thinking there should be some testcases to test whether the feature evaluation process will produce the correct variant assignment reason.